var/spack/repos/builtin/packages/fontconfig/package.py

```diff
@@ -23,6 +23,16 @@ class Fontconfig(AutotoolsPackage):
     depends_on('font-util')
     depends_on('libuuid', when='@2.13.1:')
 
+    def setup_build_environment(self, spack_env):
+        spack_env.set(
+            "FREETYPE_CFLAGS",
+            "-I{0}/include/freetype2".format(
+                self.spec['freetype'].prefix))
```

Comment:
It would be better to use `spec['freetype'].headers.include_flags`

Reply:
Hm.. that gives me the -I.../include but not the -I../include/freetype2, and the build blows up...  maybe if I scribble the /freetype2 on the end(?)...

Reply:
Okay that works, and is somewhat prettier.  Just a note, other than the spack-nsf-doe-hpc-tutorial slides, I don't see the spec.lib.ld_flags or spec.header.include_flags usage documented anywhere...

Reply:
> Hm.. that gives me the -I.../include but not the -I../include/freetype2, and the build blows up... maybe if I scribble the /freetype2 on the end(?)...

You may need to override the `headers` attribute of `freetype`

> Just a note, other than the spack-nsf-doe-hpc-tutorial slides, I don't see the spec.lib.ld_flags or spec.header.include_flags usage documented anywhere...

It's slightly documented in https://spack-tutorial.readthedocs.io/en/latest/tutorial_advanced_packaging.html#retrieving-library-information and we have more thorough API docs in https://spack.readthedocs.io/en/latest/llnl.util.html#module-llnl.util.filesystem